File: can_sum.py
def canSum(target, numbers):
    if (target == 0):
        return True
    if (target < 0):
        return False

    for number in numbers:
        if result := canSum(target - number, numbers):
            return True
    return False

print(canSum(7, [2, 4]))
print(canSum(7, [2, 3, 4, 7]))
print(canSum(100, [2, 3]))

File: can_sum.cpp
#include <vector>
#include <map>
using namespace std;

bool canSum (int target, vector<int> numbers, map<int, bool> memo = map<int, bool>()) {
	if (target == 0)
		return true;
	if (target < 0)
		return false;
	if (memo[target])
		return memo[target];

	for (int number : numbers) {
		memo[target - number] = canSum(target - number, numbers);
		if (memo[target - number])
			return true;
	}
	return false;
}

int main (int argc, char** argv) {
	printf("%d\n", canSum(7, vector<int>(2, 4)));
	printf("%d\n", canSum(7, vector<int>{2, 4, 3, 7}));
	printf("%d\n", canSum(300, vector<int>{7, 14}));
	return 0;
}
